worker/backend/backend_test.go

```diff
@@ -174,6 +177,226 @@ func (s *BackendSuite) TestContainersConversion() {
 	s.Len(containers, 2)
 }
 
+func (s *BackendSuite) TestLookupEmptyHandleError() {
+	_, err := s.backend.Lookup("")
+	s.Equal(err, backend.InputValidationError{})
+}
+
+func (s *BackendSuite) TestLookupCallGetContainerWithHandle() {
+	_, _ = s.backend.Lookup("handle")
+	s.Equal(1, s.client.GetContainerCallCount())
+
+	_, handle := s.client.GetContainerArgsForCall(0)
+	s.Equal("handle", handle)
+}
+
+func (s *BackendSuite) TestLookupCallGetContainerWithNamespace() {
+	_, _ = s.backend.Lookup("handle")
+	s.Equal(1, s.client.GetContainerCallCount())
+
+	ctx, _ := s.client.GetContainerArgsForCall(0)
+	s.NotNil(ctx)
+
+	namespace, found := namespaces.Namespace(ctx)
+	s.True(found)
+	s.Equal(testNamespace, namespace)
+}
+
+func (s *BackendSuite) TestLookupGetContainerError() {
+	s.client.GetContainerReturns(nil, errors.New("containerd-err"))
+
+	_, err := s.backend.Lookup("handle")
+	s.Error(err)
+	s.Equal(err, errors.New("containerd-err"))
+}
+
+func (s *BackendSuite) TestLookupGetContainerFails() {
+	s.client.GetContainerReturns(nil, errors.New("err"))
+	_, err := s.backend.Lookup("non-existent-handle")
+	s.Error(err)
+	s.Equal(err, errors.New("err"))
+}
+
+func (s *BackendSuite) TestLookupGetContainer() {
+	s.client.GetContainerReturns(new(libcontainerdfakes.FakeContainer), nil)
+	container, err := s.backend.Lookup("non-existent-handle")
```

Comment:
`non-existent-handle` could be changed to some existing container

Reply:
yes this test is bad right now -- related to the incomplete implementation of `Lookup`. Will be sorted out later 😂